components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/resources/web/entitlement/select-attribute.jsp

```diff
@@ -273,7 +273,15 @@
 
     function doCancel(){
         preSubmit();
-        document.attributeValueForm.action = "<%=returnPage%>.jsp?ruleId=" + '<%=ruleId%>';
+        <%
+        String cancelReturnPage=returnPage;
+
+        if(!returnPage.contains(".jsp")){
+           cancelReturnPage = cancelReturnPage+".jsp";
+        }
+
+        %>
+        document.attributeValueForm.action = "<%= cancelReturnPage%>?ruleId=" + '<%=ruleId%>';
```

Comment:
This doesn't fix the root of the problem. The root of the problem is allowing the user to click on "Select Resources Names" without filling in the "Rule Name". If "Rule Name" is filled in this issue doesn't occur.

Similar measure has been taken in filling in the "Entitlement Policy Name" at the top. If "Entitlement Policy Name" is not filled the "Select Resources Names" icon gives a pop up and asks the user to fill it first. Similar measure can be done for the "Rule Name" as well.